src/CashuMint.ts

```diff
@@ -19,27 +19,38 @@ import { checkResponse, checkResponseError, isObj } from './utils.js';
  * Class represents Cashu Mint API.
  */
 class CashuMint {
-	constructor(public mintUrl: string) {}
+	constructor(private _mintUrl: string) {}
+	get mintUrl() {
+		return this._mintUrl;
+	}
 
-	async getInfo(): Promise<GetInfoResponse> {
-		const { data } = await axios.get<GetInfoResponse>(`${this.mintUrl}/info`);
+	public static async getInfo(mintUrl: string): Promise<GetInfoResponse> {
+		const { data } = await axios.get<GetInfoResponse>(`${mintUrl}/info`);
 		return data;
 	}
-
-	async requestMint(amount: number): Promise<RequestMintResponse> {
-		const { data } = await axios.get<RequestMintResponse>(`${this.mintUrl}/mint`, {
+	async getInfo(): Promise<GetInfoResponse> {
+		return CashuMint.getInfo(this._mintUrl);
+	}
+	public static async requestMint(mintUrl: string, amount: number): Promise<RequestMintResponse> {
+		const { data } = await axios.get<RequestMintResponse>(`${mintUrl}/mint`, {
 			params: { amount }
 		});
 		return data;
 	}
-
-	async mint(payloads: { outputs: Array<SerializedBlindedMessage> }, hash: string) {
+	async requestMint(amount: number): Promise<RequestMintResponse> {
+		return CashuMint.requestMint(this._mintUrl, amount);
+	}
+	public static async mint(
+		mintUrl: string,
+		payloads: { outputs: Array<SerializedBlindedMessage> },
+		hash: string
+	) {
 		try {
 			const { data } = await axios.post<
 				{
 					promises: Array<SerializedBlindedSignature>;
 				} & ApiError
-			>(`${this.mintUrl}/mint`, payloads, {
+			>(`${mintUrl}/mint`, payloads, {
```

Comment:
wow the syntax with this formating confused me a lot. is there any way we can simplify the type? it's not that important if there is no simple solution.